src/main.ts

```diff
@@ -155,4 +161,5 @@ function generateHandleServerProviderEvent() {
 
 // this method is called when your extension is deactivated
 export function deactivate(): void {
+	TelemetryReporter.dispose()
```

Comment:
This should be added to the list of subscriptions in the activation (`context.subscriptions.push(TelemetryReporter)`)

Deactivate doesn't await and so it can still exit without flushing all the events, the subscription disposables are all cleaned up though so should be used for ensuring that.

Reply:
done